File: src/backend/BotBattle.Engine/Models/BoardState.cs
using BotBattle.Core;
using BotBattle.Core.Enums;
using MessagePack;

namespace BotBattle.Engine.Models;

[MessagePackObject]
public class BoardState
{
    public BoardState()
    {
    }

    public BoardState(int width, int height)
    {
        Width = width;
        Height = height;
    }

    [Key(0)]
    public int Width { get; init; }
    
    [Key(1)]
    public int Height { get; init; }
    
    [Key(2)]
    public List<Tank> Tanks { get; init; } = [];
    
    [Key(3)]
    public List<Bullet> Bullets { get; init; } = [];
    
    [Key(4)]
    public GameStatus Status { get; set; } = GameStatus.InProgress;
    
    [Key(5)]
    public List<EventLog> EventLogs { get; init; } = [];
    
    [Key(6)]
    public int Turns { get; set; } = 0;
    
    [Key(7)]
    public List<Obstacle> Obstacles { get; init; } = [];
    
    [Key(8)]
    public List<CollectibleItem> CollectibleItems { get; init; } = [];

    [Key(9)] 
    public Airplane Airplane { get; init; } = new();
}

File: src/frontend/src/lib/types/board-state.ts
import type { Airplane } from './airplane';
import type { Bullet } from './bullet';
import type { CollectibleItem } from './collectibleItem';
import type { EventLog } from './eventLog';
import type { GameStatus } from './gameStatus';
import type { Obstacle } from './obstacle';
import type { Tank } from './tank';

export type BoardState = {
	Width: number;
	Height: number;
	Tanks: Tank[];
	Bullets: Bullet[];
	Status: GameStatus;
	EventLogs: EventLog[];
	Turns: number;
	Obstacles: Obstacle[];
	CollectibleItems: CollectibleItem[];
	Airplane: Airplane;
};
